### PR TITLE
Custom OkHttpClient

### PR DESCRIPTION
**Pull request by jeffbordon**
Added ability to set custom OkHttpClient, e.g. for custom caching, authententicator, timeouts, etc.

### CODE REVIEW

**Comment by stkent** [comment]
Nice feature add! I have a couple comments I'll make inline, then we can wait for @btkelly to take a look also.

File: gandalf/build.gradle [hunk 1]
```diff
@@ -65,9 +65,9 @@ gnag {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
 
-    compile 'com.android.support:appcompat-v7:23.2.0'
+    compile 'com.android.support:appcompat-v7:24.2.1'
```
Comment (stkent) [inline gandalf/build.gradle]: In general it would be preferable to separate PRs that bump dependency versions from PRs that add new features unless the feature _requires_ that newer version :)

**Comment by stkent** [comment]
Regarding the CI failure:

```
* What went wrong:
A problem occurred configuring root project 'gandalf'.
> A problem occurred configuring project ':example'.
   > A problem occurred configuring project ':gandalf'.
      > Failed to notify project evaluation listener.
         > You have not accepted the license agreements of the following SDK components:
           [Android SDK Platform 24].
           Before building your project, you need to accept the license agreements and complete the installation of the missing components using the Android Studio SDK Manager.
           Alternatively, to learn how to transfer the license agreements from one workstation to another, go to http://d.android.com/r/studio-ui/export-licenses.html
         > Could not get unknown property 'generateReleaseJavadocJar' for object of type org.gradle.api.internal.artifacts.dsl.DefaultArtifactHandler.
```

Not sure if this is a Travis issue or our issue, @btkelly?

**Comment by stkent** [comment]
If we want to merge this without reverting the dependency updates, @jeffbordon you will need to update the `.travis.yml` file in the root of the repository so that the requested versions of the SDK components align with your changes.

**Comment by stkent** [comment]
Ah, very interesting, this probably explains why the CI build failed the way it did:

https://developer.android.com/studio/intro/update.html#download-with-gradle

**Comment by jsibbold** [comment]
add a second
-tools
line to the travis yml, also specify version 24 for the build tools etc eg:
  - tools
  - tools
  - platform-tools
  - build-tools-24.0.3
  - android-24
  - extra-android-m2repository
  - extra-android-support
  - doc-24

**Comment by jeffbordon** [comment]
Thanks for the feedback! I'm new to this and since I'm unsure about Travis config I reverted the dependency bumps. Hope that works :)

**Comment by stkent** [comment]
Solid, thanks for reverting those changes - this is a really nicely-focused change now! 💯

**Comment by stkent** [comment]
I think Bryan is on vacation all next week, but he should be able to take a look at this once he's back :)

**Comment by btkelly** [comment]
This looks good to me, nice feature!

**Comment by btkelly** [comment]
Approved
